client/components/breadcrumb/index.tsx

```diff
@@ -121,9 +121,10 @@ const Breadcrumb: React.FunctionComponent< Props > = ( props ) => {
 
 	if ( compact ) {
 		const urlBack = mobileItem?.href ?? items[ items.length - 2 ].href;
+		const onClick = mobileItem?.onClick ?? items[ items.length - 2 ].onClick;
```

Comment:
Pass the onClick to the Breadcrumb component to keep the event tracking on mobile.